src/main/java/sparksoniq/spark/iterator/flowr/WhereClauseSparkIterator.java

```diff
@@ -111,15 +112,19 @@ private void setNextLocalTupleResult() {
     public JavaRDD<FlworTuple> getRDD(DynamicContext context) {
         if (this._child != null) {
             this._rdd = _child.getRDD(context);
-            this._rdd = this._rdd.filter(new WhereClauseClosure(_expression));
-        } else {
-            throw new SparksoniqRuntimeException("Invalid where clause.");
+            this._rdd = this._rdd.filter(new OLD_WhereClauseClosure(_expression));
+            return _rdd;
         }
-        return _rdd;
+        throw new SparksoniqRuntimeException("Invalid where clause.");
```

Comment:
In this kind of situation, for cosmetics, you can negate the if condition, throw the exception if child==null and then continue with the code with one less level of nesting, which improves readability (this is a general good practice).